File: backend/src/test/java/lan/dk/podcastserver/manager/worker/noop/PassThroughExtractorTest.java
package lan.dk.podcastserver.manager.worker.noop;

import lan.dk.podcastserver.entity.Item;
import lan.dk.podcastserver.manager.downloader.DownloadingItem;
import lan.dk.podcastserver.manager.worker.noop.PassThroughExtractor;
import org.junit.Before;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static io.vavr.API.*;

/**
 * Created by kevin on 03/12/2017
 */
public class PassThroughExtractorTest {

    private PassThroughExtractor extractor;

    @Before
    public void beforeEach() {
        extractor = new PassThroughExtractor();
    }

    @Test
    public void should_return_item_and_its_url_as_is() {
        /* GIVEN */
        Item item = Item.builder()
                .url("Foo")
                .build();

        /* WHEN  */
        DownloadingItem extractedValue = extractor.extract(item);

        /* THEN  */
        assertThat(extractedValue.getItem()).isEqualTo(item);
        assertThat(extractedValue.getUrls()).containsOnly(item.getUrl());
    }

    @Test
    public void should_return_max_compatibility_minus_one() {
        assertThat(extractor.compatibility("foo")).isEqualTo(Integer.MAX_VALUE-1);
    }

}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/manager/worker/noop/PassThroughExtractorTest.kt
package com.github.davinkevin.podcastserver.manager.worker.noop

import com.github.davinkevin.podcastserver.entity.Status
import com.github.davinkevin.podcastserver.manager.downloader.DownloadingItem
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.net.URI
import java.util.*

/**
 * Created by kevin on 03/12/2017
 */
class PassThroughExtractorTest {

    private var extractor: PassThroughExtractor = PassThroughExtractor()

    @Test
    fun `should return item and its url as is`() {
        /* GIVEN */
        val item = DownloadingItem (
                id = UUID.randomUUID(),
                title = "any",
                status = Status.NOT_DOWNLOADED,
                url = URI("https://www.any.com/video/foo?param=1"),
                numberOfFail = 0,
                progression = 0,
                podcast = DownloadingItem.Podcast(
                        id = UUID.randomUUID(),
                        title = "anyPodcast"
                ),
                cover = DownloadingItem.Cover(
                        id = UUID.randomUUID(),
                        url = URI("https://anyDomain/anyPodcast/cover.jpg")
                )
        )

        /* WHEN  */
        val extractedValue = extractor.extract(item)

        /* THEN  */
        assertThat(extractedValue.item).isEqualTo(item)
        assertThat(extractedValue.urls).containsOnly(item.url.toASCIIString())
    }

    @Test
    fun `should return max compatibility minus one`() {
        assertThat(extractor.compatibility(URI("https://foo.bar.com/")))
                .isEqualTo(Integer.MAX_VALUE - 1)
    }

}
